patches/tModLoader/Terraria/ModLoader/GlobalNPC.cs

```diff
@@ -85,14 +85,22 @@ public virtual void ModifyTypeName(NPC npc, ref string typeName)
 
 	/// <summary>
 	/// Allows you to modify the bounding box for hovering over the given NPC (affects things like whether or not its name is displayed).
+	/// <br/>This hook is not called if <see cref="NPC.ShowNameOnHover"/> is set to <see langword="false">.
```

Comment:
`This hook is only called when <see cref="NPC.ShowNameOnHover"/> returns <see langword="true">`